### PR TITLE
Legacy docs

### PR DESCRIPTION
**Pull request by svlandeg**
## Description
Introducing a dedicated page in the docs to document legacy functions.
Perhaps in time we'd want a submenu and separate pages? But it seemed a little overkill for now...

Where do we want this in the sidebar? I added it to "Other" for now.

Related PR: https://github.com/explosion/spacy-legacy/pull/3

### Types of change
docs fix

## Checklist
- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by ines — APPROVED** [review]

File: website/meta/sidebars.json [hunk 1]
```diff
@@ -122,7 +122,8 @@
                     { "text": "Scorer", "url": "/api/scorer" },
                     { "text": "StringStore", "url": "/api/stringstore" },
                     { "text": "Vectors", "url": "/api/vectors" },
-                    { "text": "Vocab", "url": "/api/vocab" }
+                    { "text": "Vocab", "url": "/api/vocab" },
+                    { "text": "Legacy", "url": "/api/legacy" }
                 ]
```
Comment (adrianeboyd) [inline website/meta/sidebars.json:127]: This list is otherwise alphabetized so this is a bit weird. Maybe it could go at the end of "Overview" instead?

Comment (svlandeg) [reply website/meta/sidebars.json:127]: It feels perhaps a bit too prominent in Overview? And it'd push down other more import parts :(

Comment (svlandeg) [reply website/meta/sidebars.json:127]: We could also have an entire separate section all the way at the bottom (perhaps before CYTHON), assuming we'll be subdividing this at some point anyway...

Comment (adrianeboyd) [reply website/meta/sidebars.json:127]: I think a separate section sounds good.

**Pull request merged**